src/parser/rogue/outlaw/modules/spells/shadowlands/legendaries/GuileCharm.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import Analyzer from 'parser/core/Analyzer';
+import Abilities from 'parser/core/modules/Abilities';
+import SPELLS from 'common/SPELLS';
+import Statistic from 'interface/statistics/Statistic';
+import { formatPercentage } from 'common/format';
+import BoringSpellValue from 'interface/statistics/components/BoringSpellValue';
+import SpellLink from 'common/SpellLink';
+
+class GuileCharm extends Analyzer {
+  static dependencies = {
+    abilities: Abilities,
+  };
+
+  deepInsightUptime: number = 0;
+  moderateInsightUptime: number = 0;
+  shallowInsightUptime: number = 0;
+  protected abilities!: Abilities;
+
+  constructor(options: any) {
+    super(options);
+    this.active = this.selectedCombatant.hasLegendaryByBonusID(SPELLS.GUILE_CHARM.bonusID);
+  }
+
+  get percentUptime() {
+    this.shallowInsightUptime = this.selectedCombatant.getBuffUptime(SPELLS.SHALLOW_INSIGHT_BUFF.id) / this.owner.fightDuration;
+    this.moderateInsightUptime = this.selectedCombatant.getBuffUptime(SPELLS.MODERATE_INSIGHT_BUFF.id) / this.owner.fightDuration;
+    this.deepInsightUptime = this.selectedCombatant.getBuffUptime(SPELLS.DEEP_INSIGHT_BUFF.id) / this.owner.fightDuration;
+    return {
+      shallowInsight: this.shallowInsightUptime,
+      moderateInsight: this.moderateInsightUptime,
+      deepInsight: this.deepInsightUptime,
+    };
+  }
+
+  statistic() {
+    return (
+      <>
+        <Statistic size="flexible" tooltip="This measures how long each buff from this legendary is active.">
+          <div className="pad">
+            <label><SpellLink id={SPELLS.GUILE_CHARM.id} /> Insight Uptime</label>
+          </div>
+          <BoringSpellValue spell={SPELLS.SHALLOW_INSIGHT_BUFF} label={''} value={`${formatPercentage(this.percentUptime.shallowInsight)}%`} />
```

Comment:
Also not sure what having multiple BoringSpellValue things does, but a screenshot of the boxes should help

Reply:
![image](https://user-images.githubusercontent.com/23483002/95269719-a9196200-07ff-11eb-9a74-ddb30c33403c.png)
This is what it will output. I'm not 100% on the design of it. But I wanted to get something out there and get feedback on it.

Reply:
```
statistic() {
    return (
      <>
        <Statistic 
          size="flexible" 
          tooltip="This measures how long each buff from this legendary is active."
        >
          <BoringSpellValueText spell={SPELLS.GUILE_CHARM}>
            <SpellIcon id={SPELLS.SHALLOW_INSIGHT_BUFF.id} /> {formatPercentage(this.percentUptime.shallowInsight)}% <small>Shallow Insight uptime</small><br />
            <SpellIcon id={SPELLS.MODERATE_INSIGHT_BUFF.id} /> {formatPercentage(this.percentUptime.moderateInsight)}% <small>Moderate Insight uptime</small><br />
            <SpellIcon id={SPELLS.DEEP_INSIGHT_BUFF.id} /> {formatPercentage(this.percentUptime.deepInsight)}% <small>Deep Insight uptime</small>
          </BoringSpellValueText>
        </Statistic>
      </>
    );
  }
```
You can do something like that so there isnt a ton of empty space everywhere
![image](https://user-images.githubusercontent.com/1304554/95284882-0542ad00-0825-11eb-9194-e0c00165bbd1.png)

Reply:
I like that much better! I'll implement that later this morning.